library/ssl_tls.c

```diff
@@ -3102,6 +3102,27 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
     }
 #endif
 
+#if !defined(DEPRECATED_REMOVED)
+    /* Heap allocate and translate curve_list from internal to IANA group ids */
+    if ( ssl->conf->curve_list ) {
+        size_t length;
+	for (length = 0; ssl->conf->curve_list[length] != MBEDTLS_ECP_DP_NONE; length++) {} 
```

Comment:
Maybe pick _some_ upper bound here to avoid looping until crashing if the user forgets to 0-terminate the list.